src/actions/actions.ts

```diff
@@ -1739,6 +1781,10 @@ class CommandSearchCurrentWordExactForward extends BaseCommand {
   runsOnceForEachCountPrefix = true;
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    if (Configuration.visualstar && vimState.currentMode !== ModeName.Normal) {
```

Comment:
Instead of a custom dispatch on current mode, can we have two separate commands, one for ModeName.Normal, and one for ModeName.Visual/Line?